### PR TITLE
Added limit of items per tile

### PR DESCRIPTION
**Pull request by EPuncker**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Added configurable limit of items per tile exactly as it was on 0.x series!

If you have any change suggestion, please use the _**inline code suggestion**_ feature for faster changes

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Comment by nekiro** [comment]
This is easily doable, just with onMoveItem event in lua

**Review by DSpeichert — COMMENTED** [review]
LGTM, although I'm not sure if it should be in Lua instead or not. Looks quite elegant here. Also - do we need it?

**Comment by MillhioreBT** [comment]
This could be in a lua script, for example:

**data/scripts/eventcallbacks/tile_limits.lua**
```lua
local tileLimit = 0
local protectionTileLimit = 0
local houseTileLimit = 0

local ec = EventCallback

function ec.onMoveItem(player, item, count, fromPosition, toPosition, fromCylinder, toCylinder)
	local tile = toPosition:getTile()
	if tile then
		local itemLimit = tile:getHouse() and houseTileLimit or tile:hasFlag(TILESTATE_PROTECTIONZONE) and protectionTileLimit or tileLimit
		if itemLimit > 0 and tile:getThingCount() > itemLimit and item:getType():getType() ~= ITEM_TYPE_MAGICFIELD then
			player:sendCancelMessage("You can not add more items on this tile.")
			return false
		end
	end
	return true
end

ec:register(-1)
```

However your code in the sources is correct, I will check it thoroughly later sure ;)

File: src/tools.cpp [hunk 1]
```diff
@@ -1228,6 +1228,9 @@ const char* getReturnMessage(ReturnValue value)
 		case RETURNVALUE_ITEMCANNOTBEMOVEDTHERE:
 			return "This item cannot be moved there.";
 
+		case RETURNVALUE_YOUCANNOTADDMOREITEMSONTHISTILE:
+			return "You can not add more items on this tile.";
```
Comment (idontreallywolf) [inline src/tools.cpp:1232]:
```suggestion
			return "You cannot add more items on this tile.";
```

Comment (nekiro) [reply src/tools.cpp:1232]: why?

**Comment by EPuncker** [comment]
@nekiro @DSpeichert we don't need it, but it wouldn't harm as it is only a few lines, and yes there were a person asking for it in the forum, also after taking a look at 0.4 config.lua it was like the only "useful" setting they had...

and yes, I was not sure if C++ or Lua, but after thinking Lua is way simpler and one could even remove it totally, so I'm doing a new commit with Lua change

**Comment by EPuncker** [comment]
okay, better thinking, perhaps we should just release it in the forum instead

**Pull request closed without merging**